src/main/java/com/dotcms/saml/service/impl/DefaultMetaDescriptorServiceImpl.java

```diff
@@ -139,29 +158,30 @@ public EntityDescriptor getServiceProviderEntityDescriptor(final IdpConfig idpCo
 		final SAMLObjectBuilder<SingleLogoutService> singleLogoutServiceBuilder = (SAMLObjectBuilder<SingleLogoutService>) this.xmlObjectBuilderFactory
 				.getBuilder(SingleLogoutService.DEFAULT_ELEMENT_NAME);
 
-		final EntityDescriptor descriptor = entityDescriptorBuilder.buildObject();
+		final EntityDescriptor descriptor     = entityDescriptorBuilder.buildObject();
 		final SPSSODescriptor spssoDescriptor = spssoDescriptorBuilder.buildObject();
 
-		descriptor.setEntityID(SamlUtils.getSPIssuerValue(idpConfig)); // this
-																		// get
-																		// from
-																		// the
-																		// dotmarketing-config.properties.
+		descriptor.setEntityID(this.samlService.getSPIssuerValue(identityProviderConfiguration));
```

Comment:
@jdotcms What about adding some verifications here to see if those objects are null? Just like we saw in today's call. Some null checks and logging would help locate the problem faster.